robot_smach_states/src/robot_smach_states/navigation/follow_operator.py

```diff
@@ -86,26 +87,28 @@ def _operator_standing_still_for_x_seconds(self, timeout):
 
     def _standing_still_for_x_seconds(self, timeout):
         current_pose_stamped = self._robot.base.get_location()
+        now = rospy.Time.now()
 
         if not self._last_pose_stamped:
             self._last_pose_stamped = current_pose_stamped
+            self._last_pose_stamped_time = now
         else:
-            current_yaw = transformations.euler_z_from_quaternion(current_pose_stamped.pose.orientation)
-            last_yaw = transformations.euler_z_from_quaternion(self._last_pose_stamped.pose.orientation)
+            current_yaw = current_pose_stamped.M.GetRPY()[2]  # Get the Yaw
+            last_yaw = self._last_pose_stamped.M.GetRPY()[2]  # Get the Yaw
 
             # Compare the pose with the last pose and update if difference is larger than x
-            if math.hypot(current_pose_stamped.pose.position.x - self._last_pose_stamped.pose.position.x,
-                          current_pose_stamped.pose.position.y - self._last_pose_stamped.pose.position.y) > 0.05 or abs(current_yaw - last_yaw) > 0.3:
+            if (current_pose_stamped - self._last_pose_stamped).Norm() > 0.05 or abs(current_yaw - last_yaw) > 0.3:
```

Comment:
Does this work? You can't subtract frames directly either. You can use kdl.diff(...), which results in a kdl twist. On a Twist you can compute the norm of the vel and the rot. I guess, however, you want to subtract the positions instead of the poses